From 5ae47e9173931fedd6b3adb04ff94b772587fa2a Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@samba.org>
Date: Fri, 23 Oct 2015 15:28:23 +0200
Subject: [PATCH] nss_wins: Use lp_global_no_reinit()

This avoids that we run into use after free issues when we access memory
allocated on the globals and the global being reinitialized.

BUG: https://bugzilla.samba.org/show_bug.cgi?id=11563

Signed-off-by: Andreas Schneider <asn@samba.org>
Reviewed-by: Volker Lendecke <vl@samba.org>

(cherry picked from commit 0abbfb2e4d5bcd847983ef7981840f1eab7b917c)
---
 nsswitch/wins.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nsswitch/wins.c b/nsswitch/wins.c
index d63968b..6b5c3e2 100644
--- a/nsswitch/wins.c
+++ b/nsswitch/wins.c
@@ -53,7 +53,7 @@ static void nss_wins_init(void)
 
 	TimeInit();
 	setup_logging("nss_wins",False);
-	lp_load(get_dyn_CONFIGFILE(),True,False,False,True);
+	lp_load_global_no_reinit(get_dyn_CONFIGFILE());
 	load_interfaces();
 }
 
-- 
2.6.2

